src/OpenLoco/src/Vehicles/VehicleManager.cpp

```diff
@@ -135,9 +135,36 @@ namespace OpenLoco::VehicleManager
     // 0x004B93DC
     void resetIfHeadingForStation(const StationId stationId)
     {
-        registers regs;
-        regs.ebx = enumValue(stationId);
-        call(0x004B93DC, regs);
+        for (auto* v : VehicleList())
+        {
+            Vehicles::Vehicle train(*v);
+            // Stop train from heading to the station
+            if (train.head->stationId == stationId)
+            {
+                train.head->stationId = StationId::null;
+            }
+
+            // Remove any station references from cargo
+            // NOTE: Deletes the cargo!
+            for (auto& car : train.cars)
+            {
+                for (auto& carComponent : car)
+                {
+                    if (carComponent.front->secondaryCargo.qty != 0 && carComponent.front->secondaryCargo.townFrom == stationId)
```

Comment:
you don't need any of these checks for `!= 0`. Regardless of the value of `qty`, the new value will always be zero (assuming the second check passes)

Reply:
good point.